ArduCopter/mode_guided.cpp

```diff
@@ -326,6 +326,7 @@ bool ModeGuided::set_destination(const Vector3f& destination, bool use_yaw, floa
     const Location dest_loc(destination, terrain_alt ? Location::AltFrame::ABOVE_TERRAIN : Location::AltFrame::ABOVE_ORIGIN);
     if (!copter.fence.check_destination_within_fence(dest_loc)) {
         AP::logger().Write_Error(LogErrorSubsystem::NAVIGATION, LogErrorCode::DEST_OUTSIDE_FENCE);
+        gcs().send_text(MAV_SEVERITY_NOTICE, "target violates fence");
         // failure is propagated to GCS with NAK
```

Comment:
If the comment is correct the GCS should be able to report a failure. There is more than one path where this could fail tho.

Reply:
Hi @IamPete1 

I agree with you and I made some changes. 
MAV_CMD_DO_REPOSITION will report failure so we do not brother sending text
https://github.com/ArduPilot/ardupilot/blob/7e07859ec93b1643d335de01bcc7bbe5bd655ab1/ArduCopter/GCS_Mavlink.cpp#L695-L697
but SET_POSITION_TARGET_GLOBAL_INT will not report failure so we send text to notify users
https://github.com/ArduPilot/ardupilot/blob/7e07859ec93b1643d335de01bcc7bbe5bd655ab1/ArduCopter/GCS_Mavlink.cpp#L1284-L1286